ref_builder/ncbi/client.py

```diff
@@ -40,6 +43,26 @@ class GenbankRecordKey(StrEnum):
     SEQUENCE = "GBSeq_sequence"
 
 
+@dataclass
```

Comment:
I think this configuration should just be kwargs at this point. If the number of fields grows in the future, a separate class is a good idea.

Here's why:
* No need to document another class. It is currently undocumented and untested.
* No need to name another class. The current name verges on a leaky abstraction.
* Put code for composing string next to the code that builds the rest of the `term`, avoiding indirection.